test/dynamo/test_export.py

```diff
@@ -1752,6 +1752,94 @@ def fn_with_kwargs(pos0, tuple0, *myargs):
         self.assertTrue(torch._dynamo.utils.same(real_result, dynamo_result))
 
 
+    def test_export_trainig(self):
+        class Module(torch.nn.Module):
+            def __init__(self):
+                super().__init__()
+                self.linear = torch.nn.Linear(3, 3)
+
+            def forward(self, val):
+                out = self.linear(val)
+                out.relu_()
+                loss = out.sum()
+                return loss
+
+        mod = Module()
+        x = torch.randn([3, 3], requires_grad=True)
+
+        torch._dynamo.reset()
+        joint_gm, _  = torch._dynamo.export(mod, x, aten_graph=True, training_mode='joint', tracing_mode='symbolic')
+
+        self.assertExpectedInline(joint_gm.print_readable(), """\
```

Comment:
Doing tests this way will be brittle as if we change our printing logic or variable naming logic or some small part of the trace they will fail.

Could we instead test the functionality of the graph?

Reply:
So, the point of expect tests is that they're easy to update (you just run with `EXPECTTEST_ACCEPT=1` and the source files are all updated with the new values). That being said, we can't land *these* tests, they have absolute filepaths hardcoded in them; they'll have to be normalized at least.

Reply:
I don't plan to check this test in. Just showing the existing exported graph to discuss what are the changes we should make. In particular, we need to design the properties, invariants, conventions of exported training graphs.